### PR TITLE
Mobile: BottomSheet design tweaks v2

### PR DESCRIPTION
**Pull request by etoledom**
## Description

This PR improves the design of the BottomSheet following https://github.com/wordpress-mobile/gutenberg-mobile/issues/535 .
Also added fixed for:
https://github.com/wordpress-mobile/gutenberg-mobile/issues/597
https://github.com/wordpress-mobile/gutenberg-mobile/issues/596

- Added more space at the top of the BottomSheet (it should sum up to 24)
- Make last row’s border-bottom full-width.
  - Added separator styles: `fullWidth`, `leftMargin`, and `none`. If it's not specified, it will behave as before.
  - @marecar3 - This change might hit the BottomSheet on your PR too.
- Truncate long text strings in the middle instead of at the end of the line.
  - This one was tricky since TextInput doesn't have this configuration. I ended up replacing the TextInput with a Text component to display the value, and change it back to TextInput when the user want's to edit it.
- Force title case for the row titles.
  - We don't have a "native" way of doing it in RN (not supported), so I opted for doing it manually for now.

bottom-sheet-design

**To test:**
- Check out the related `gutenberg-mobile` PR (https://github.com/wordpress-mobile/gutenberg-mobile/pull/594).
- Display the Image Settings BottomSheet.
- Check that the placeholders are displaying properly.
- Check that you can edit the settings by pressing on the cells.
- Check that the design looks good 😎 

Swipe:
- Test swipe to dismiss from anywhere in the BottomSheet.

### CODE REVIEW

**Comment by etoledom** [comment]
Hey @marecar3 ! I added some extra commits to implement the picker design by Thomas. This is how it looks now on Android (while iOS has no visual changes):

android

By default those changes fixed https://github.com/wordpress-mobile/gutenberg-mobile/issues/597

And I also took the opportunity to fix:
https://github.com/wordpress-mobile/gutenberg-mobile/issues/596

I updated the description with a new test step.

**Comment by marecar3** [comment]
Thanks for the update @etoledom. I will take a look :)

**Comment by iamthomasbishop** [comment]
@etoledom One thing I just thought about regarding the case where we're using the Bottom Sheet component as a replacement/equivalent to iOS `ActionSheets` – we might want to consider removing the drag handle, as there is no real need to have the handle for this type of quick item selection interaction.

Visual example: 

image

* 1: full sheet
* 2, 3: "Action" Sheets

Something to consider moving forward.

File: packages/editor/src/components/mobile/bottom-sheet/cell.native.js [hunk 1]
```diff
@@ -7,85 +7,132 @@ import { TouchableOpacity, Text, View, TextInput, I18nManager } from 'react-nati
...
-		valueStyle = {},
-		onChangeValue,
-		children,
-		editable = true,
-		...valueProps
-	} = props;
+export default class Cell extends Component {
+	constructor() {
+		super( ...arguments );
+		this._valueTextInput = undefined;
```
Comment (marecar3) [inline packages/editor/src/components/mobile/bottom-sheet/cell.native.js]: Maybe if we already know that this. _valueTextInput is undefined by default, we don't need to assign it to undefined?

File: packages/editor/src/components/mobile/bottom-sheet/cell.native.js [hunk 2]
```diff
@@ -7,85 +7,132 @@ import { TouchableOpacity, Text, View, TextInput, I18nManager } from 'react-nati
  * WordPress dependencies
  */
 import { Dashicon } from '@wordpress/components';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import styles from './styles.scss';
+import platformStyles from './cellStyles.scss';
```
Comment (marecar3) [inline packages/editor/src/components/mobile/bottom-sheet/cell.native.js]: @etoledom I have added here `.scss` so that test can pass.

**Comment by etoledom** [comment]
Thank you @marecar3 for the review and for the fix!
I have updated `gutenberg-mobile` and it should be passing now 🙏

File: packages/editor/src/components/mobile/bottom-sheet/cell.native.js (hunk 2 above)
Comment (etoledom) [reply packages/editor/src/components/mobile/bottom-sheet/cell.native.js]: Thank you!

**Review by marecar3 — APPROVED** [review]
LGTM!

**Pull request merged**